--- ./libntp/machines.c	Mon Nov  8 04:29:32 1999
+++ ./libntp/machines.c.sz	Sun Apr  1 14:08:28 2001
@@ -4,6 +4,8 @@
  *
  */
 
+#include <time.h>
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
--- ./ntpd/ntp_timer.c	Tue Jul  4 20:45:35 2000
+++ ./ntpd/ntp_timer.c.sz	Sun Apr  1 14:10:28 2001
@@ -1,6 +1,8 @@
 /*
  * ntp_timer.c - event timer support routines
  */
+#include <time.h>
+
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
--- ./ntpdate/ntpdate.c	Wed May  3 07:25:26 2000
+++ ./ntpdate/ntpdate.c.sz	Sun Apr  1 14:10:47 2001
@@ -2,6 +2,8 @@
  * ntpdate - set the time of day by polling one or more NTP servers
  */
 
+#include <time.h>
+
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
--- ./ntpdate/ntptimeset.c	Sun May 14 02:53:57 2000
+++ ./ntpdate/ntptimeset.c.sz	Sun Apr  1 14:11:39 2001
@@ -117,6 +117,8 @@
  *
  */
 
+#include <time.h>
+
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
