diff -ur ntp-4.1.1a.orig/configure.ac ntp-4.1.1a/configure.ac
--- ntp-4.1.1a.orig/configure.ac	Thu Mar 28 23:42:00 2002
+++ ntp-4.1.1a/configure.ac	Thu Nov 21 11:56:41 2002
@@ -687,9 +687,9 @@
  yes)
     ;;
  *)
-    AC_CHECK_LIB([elf], [nlist])	dnl Only needed for tickadj...
+    #AC_CHECK_LIB([elf], [nlist])	dnl Only needed for tickadj...
     dnl AC_CHECK_LIB(kvm, main, , , -lelf)
-    AC_CHECK_LIB([kvm], [main])		dnl We already know about -lelf here...
+    #AC_CHECK_LIB([kvm], [main])		dnl We already know about -lelf here...
     AC_CHECK_LIB([ld], [nlist])
     AC_CHECK_LIB([mld], [nlist])
     AC_CHECK_HEADERS([nlist.h sys/var.h])
