--- ntp-4.2.8/configure.ac~	2014-12-19 12:56:53.000000000 +0100
+++ ntp-4.2.8/configure.ac	2014-12-19 17:04:09.696796977 +0100
@@ -422,9 +422,9 @@
     # nlist stuff is only needed for tickadj.
     saved_LIBS="$LIBS"
     LIBS=
-    AC_SEARCH_LIBS([nlist], [elf ld mld])
+    #AC_SEARCH_LIBS([nlist], [elf ld mld])
     # XXX ac_cv_search_nlist will be 'none required', 'no', or '-l...'
-    AC_SEARCH_LIBS([kvm_open], [kvm])	dnl We already know about -lelf here...
+    #AC_SEARCH_LIBS([kvm_open], [kvm])	dnl We already know about -lelf here...
     # XXX ac_cv_search_kvm_open will be 'none required', 'no', or '-l...'
     AC_CHECK_HEADERS([nlist.h sys/var.h])
     case "$ac_cv_header_nlist_h" in
