diff -urNp -x '*.orig' ntp-4.2.8p12.org/ntpd/ntp_io.c ntp-4.2.8p12/ntpd/ntp_io.c
--- ntp-4.2.8p12.org/ntpd/ntp_io.c	2021-10-15 23:43:02.012145954 +0200
+++ ntp-4.2.8p12/ntpd/ntp_io.c	2021-10-15 23:43:02.175478857 +0200
@@ -4832,10 +4832,7 @@ init_async_notifications()
 #ifdef HAVE_RTNETLINK
 	ZERO(sa);
 	sa.nl_family = PF_NETLINK;
-	sa.nl_groups = RTMGRP_LINK | RTMGRP_IPV4_IFADDR
-		       | RTMGRP_IPV6_IFADDR | RTMGRP_IPV4_ROUTE
-		       | RTMGRP_IPV4_MROUTE | RTMGRP_IPV6_ROUTE
-		       | RTMGRP_IPV6_MROUTE;
+	sa.nl_groups = RTMGRP_IPV4_IFADDR | RTMGRP_IPV6_IFADDR;
 	if (bind(fd, (struct sockaddr *)&sa, sizeof(sa)) < 0) {
 		msyslog(LOG_ERR,
 			"bind failed on routing socket (%m) - using polled interface update");
