--- ntp-4.0.99k/ntpd/refclock_true.c.org	Thu Apr  5 16:08:07 2001
+++ ntp-4.0.99k/ntpd/refclock_true.c	Thu Apr  5 16:10:47 2001
@@ -212,10 +212,12 @@
 	{
 		if (want_debugging) {
 		    char filename[20];
+		    int fd;
 
 		    sprintf(filename, "/tmp/true%d.debug", up->unit);
-		    up->debug = fopen(filename, "w");
-		    if (up->debug) {
+		    fd = open(filename, O_CREAT | O_WRONLY | O_EXCL, 0600);
+		    up->debug = fdopen(fd, "w");
+		    if (fd >= 0 && up->debug) {
 #ifdef HAVE_SETVBUF
 			    static char buf[BUFSIZ];
 			    setvbuf(up->debug, buf, _IOLBF, BUFSIZ);
