--- ntp-4.2.6p1/ntpdc/Makefile.am.orig	2010-01-24 11:01:48.000000000 +0100
+++ ntp-4.2.6p1/ntpdc/Makefile.am	2010-05-24 20:59:08.000000000 +0200
@@ -14,8 +14,8 @@
 	-I$(top_srcdir)/lib/isc/nothreads/include \
 	-I$(top_srcdir)/lib/isc/unix/include $(LIBOPTS_CFLAGS)
 # LDADD might need RESLIB and ADJLIB
-ntpdc_LDADD=	version.o @EDITLINE_LIBS@ $(LIBOPTS_LDADD) \
-		../libntp/libntp.a @LCRYPTO@
+ntpdc_LDADD=	version.o $(LIBOPTS_LDADD) \
+		../libntp/libntp.a @LCRYPTO@ @EDITLINE_LIBS@
 # ntpdc-layout doesn't need any additional libraries at all
 ntpdc_layout_LDADD=
 DISTCLEANFILES=	.version version.c
--- ntp-4.2.6p1/ntpq/Makefile.am~	2010-01-24 11:01:47.000000000 +0100
+++ ntp-4.2.6p1/ntpq/Makefile.am	2010-05-24 21:06:30.000000000 +0200
@@ -12,8 +12,8 @@
 	-I$(top_srcdir)/lib/isc/unix/include $(LIBOPTS_CFLAGS)
 
 # LDADD might need RESLIB and ADJLIB
-ntpq_LDADD=	version.o @EDITLINE_LIBS@ $(LIBOPTS_LDADD) \
-		../libntp/libntp.a @LCRYPTO@
+ntpq_LDADD=	version.o $(LIBOPTS_LDADD) \
+		../libntp/libntp.a @EDITLINE_LIBS@ @LCRYPTO@
 noinst_HEADERS=	ntpq.h
 noinst_LIBRARIES=	libntpq.a
 libntpq_a_CFLAGS=	-DNO_MAIN_ALLOWED -DBUILD_AS_LIB
