--- configure.ac.org	2007-07-04 23:13:54.063029216 +0200
+++ configure.ac	2007-07-04 23:17:39.853944164 +0200
@@ -1012,7 +1012,9 @@
 	# this conditional will need to change.  Maybe use AC_TRY_RUN
 	# instead to try to set the time to itself and check errno.
     ;;
- *) AC_CHECK_FUNCS(clock_gettime clock_settime)
+ *) 
+ 	AC_CHECK_LIB(rt, clock_settime, [LIBS="$LIBS -lrt"])
+ 	AC_CHECK_FUNCS(clock_gettime clock_settime)
     ;;
 esac
 AC_CHECK_FUNCS(daemon)
