--- ntp-4.2.8p15/libntp/work_thread.c~	2018-08-14 13:51:06.000000000 +0200
+++ ntp-4.2.8p15/libntp/work_thread.c	2021-10-16 21:40:00.752349056 +0200
@@ -41,12 +41,6 @@
 #ifndef THREAD_MINSTACKSIZE
 # define THREAD_MINSTACKSIZE	(64U * 1024)
 #endif
-#ifndef __sun
-#if defined(PTHREAD_STACK_MIN) && THREAD_MINSTACKSIZE < PTHREAD_STACK_MIN
-# undef THREAD_MINSTACKSIZE
-# define THREAD_MINSTACKSIZE PTHREAD_STACK_MIN
-#endif
-#endif
 
 #ifndef THREAD_MAXSTACKSIZE
 # define THREAD_MAXSTACKSIZE	(256U * 1024)
@@ -594,8 +588,11 @@
 			"start_blocking_thread: pthread_attr_getstacksize() -> %s",
 			strerror(rc));
 	} else {
-		if (ostacksize < THREAD_MINSTACKSIZE)
-			nstacksize = THREAD_MINSTACKSIZE;
+		int minstack = THREAD_MINSTACKSIZE;
+		if (minstack < PTHREAD_STACK_MIN)
+			minstack = PTHREAD_STACK_MIN;
+		if (ostacksize < minstack)
+			nstacksize = minstack;
 		else if (ostacksize > THREAD_MAXSTACKSIZE)
 			nstacksize = THREAD_MAXSTACKSIZE;
 		else
